WordPressUI/Extensions/UIImageView+Networking.swift

```diff
@@ -41,25 +41,30 @@ public extension UIImageView {
             return
         }
 
+        let internalOnSuccess = { [weak self] (image: UIImage) in
+            self?.image = image
+            success?(image)
+        }
+
+        let cachedImage = Downloader.cache.object(forKey: url as AnyObject) as? UIImage
+
         // If we are asking for the same URL let's just stay like we are
         guard url != downloadURL else {
             if let error = downloadTask?.error {
                 failure?(error)
```

Comment:
I just noticed that we may have another issue with the code above, If I understand well if we previsouly try to download the same image and had an error we just send a failure. This will make the code to always fail for a specific URL after the first try failed. I think we should allow the code to continue to retry the download.

On the success case you are right it should just callback the success block.

Reply:
Agreed! Is better to try again than to give up.